addon/components/basic-dropdown-content.ts

```diff
@@ -137,61 +137,65 @@ export default class BasicDropdownContent extends Component<Args> {
       this.scrollableAncestors = getScrollableAncestors(triggerElement);
     }
     this.addScrollHandling(dropdownElement);
-  }
-
-  @action
-  teardown(): void {
-    this.removeGlobalEvents();
-    this.removeScrollHandling();
-    this.scrollableAncestors = [];
-
-    document.removeEventListener(
-      this.args.rootEventType,
-      this.handleRootMouseDown as RootMouseDownHandler,
-      true
-    );
+    return () => {
+      this.removeGlobalEvents();
+      this.removeScrollHandling();
+      this.scrollableAncestors = [];
 
-    if (this.isTouchDevice) {
-      document.removeEventListener('touchstart', this.touchStartHandler, true);
       document.removeEventListener(
-        'touchend',
+        this.args.rootEventType,
         this.handleRootMouseDown as RootMouseDownHandler,
         true
       );
+
+      if (this.isTouchDevice) {
+        document.removeEventListener('touchstart', this.touchStartHandler, true);
+        document.removeEventListener(
+          'touchend',
+          this.handleRootMouseDown as RootMouseDownHandler,
+          true
+        );
+      }
     }
-  }
+  // @ts-ignore
+  }, { eager: false });
```

Comment:
All `modifier()` invocations must include `{ eager: false }` since this package supports `ember-modifier` v3 and v4.

For apps using v3.2.x `{ eager: false }` will ensure the default autotracking behaviour (eager).

For apps using v4 this argument will be ignored.